### DISCUSSION TITLE
Looking for union(-like) structure

### DISCUSSION THREAD

**Discussion by ikabadzhov** [nlohmann/json#3815] [Q&A] [answered]
Hello, I was trying to parse json key-value pairs, where the values are either a single string or a vector of strings, i.e.:
in some files I have `{"name":"John"}`, but in other files I have `{"name":["John", "Smith"]}`. 

I would only like to pass them to a setter (for which I have overloads accepting a string and a vector of strings - my json handler would make use only of the first one though). And the member that is being set is a vector.

My naive solution is:
```cpp
myClass.SetName(jsn["name"].is_string() ? std::vector<std::string>{std::string{jsn["name"]}} : std::vector<std::string>{jsn["name"]})
```
I was wondering if there is more idiomatic way to do that? Or am I missing something obvious?

**Comment by gregmarr** [comment, marked as answer]
Do you NEED to only use the first?  If not, you can do this:

```
auto &name = jsn["name"];

if (name.is_string())
    myClass.SetName(name.get<std::string>());
else
    myClass.SetName(name.get<std::vector<std::string>>());
```
